stm/asyncTaskQueue.mli

```diff
@@ -6,79 +6,129 @@
 (*         *       GNU Lesser General Public License Version 2.1        *)
 (************************************************************************)
 
+(** A status is
+    - `Fresh when XXX
+    - `Old x when XXX
+  *)
 type 'a worker_status = [ `Fresh | `Old of 'a ]
 
+(** The [Task] module type defines an abstract message-processing
+    entity. *)
 module type Task = sig
 
+  (** Identifier of a task, for workers that can process several
+      jobs. Usually an integer or any other unique identifier. *)
   type task
+
+  (** XXX No idea what this does *)
   type competence
 
-  (* Marshallable *)
+  (** Type of input and output data for workers.
+
+      The data must be marshallable as it send through the network
+      using [Mashal] . *)
   type request
   type response
 
-  val name : string ref (* UID of the task kind, for -toploop *)
+  (** UID of the task kind, for -toploop *)
+  val name : string ref
+  (** Extra arguments of the task kind, for -toploop *)
   val extra_env : unit -> string array
 
-  (* run by the master, on a thread *)
+  (** {5} Client API, it is run by the master, on a thread *)
+
+  (** [request_of_task status tid] checks if the worker has already
+      processed [tid] *)
   val request_of_task : competence worker_status -> task -> request option
+
+  (** [task_match status tid] XXX *)
```

Comment:
An "old" worker can be picky and only choose tasks in the queue that match its competence.